pkg/gadgets/seccomp/tracer/bpf/seccomp.c

```diff
@@ -9,21 +9,50 @@
 #include <bpf/bpf_helpers.h>
 #include <bpf/bpf_core_read.h>
 #include <bpf/bpf_tracing.h>
+#include <bpf/bpf_tracing.h>
 
 #include "seccomp-common.h"
 
 #define TASK_COMM_LEN 16
 #define TS_COMPAT 0x0002
 
+// prctl syscall number from
+// https://github.com/seccomp/libseccomp/blob/abad8a8f41fc13efbb95fc1ccaa3e181342bade7/src/syscalls.csv#L265
+#ifndef __NR_prctl
+# if defined(bpf_target_x86)
+#  define __NR_prctl 157
+# elif defined(bpf_target_arm64)
+#  define __NR_prctl 167
+# else
+#  error "Unsupported architecture"
+# endif
+#endif
+
+// prclt syscall parameters from
+// https://github.com/torvalds/linux/blob/5147da902e0dd162c6254a61e4c57f21b60a9b1c/include/uapi/linux/prctl.h#L10
+// https://github.com/torvalds/linux/blob/5147da902e0dd162c6254a61e4c57f21b60a9b1c/include/uapi/linux/prctl.h#L175
+#ifndef PR_GET_PDEATHSIG
+# define PR_GET_PDEATHSIG 2
+#endif
+#ifndef PR_SET_NO_NEW_PRIVS
+# define PR_SET_NO_NEW_PRIVS 38
+#endif
+
 // Seccomp syscall number from
 // https://github.com/torvalds/linux/blob/v5.12/tools/testing/selftests/seccomp/seccomp_bpf.c#L115
 // Only x86_64 is supported for now.
 #ifndef __NR_seccomp
-#define __NR_seccomp 317
+# if defined(bpf_target_x86)
+#  define __NR_seccomp 317
+# elif defined(bpf_target_arm64)
+#  define __NR_seccomp 277
+# else
+#  error "Unsupported architecture"
+# endif
 #endif
 
 #ifndef SECCOMP_SET_MODE_FILTER
-#define SECCOMP_SET_MODE_FILTER         1
+# define SECCOMP_SET_MODE_FILTER 1
```

Comment:
It's not used anymore, I think we can remove it.

Reply:
Done